jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocDriver.java

```
@@ -85,8 +87,22 @@ Connection createConnection(String url, Properties info) throws SQLException {
 
         if (url.startsWith(DATAPROC_JDBC_HIVE_URL_SCHEMA)) {
             try {
-                String hiveURL = formatHiveUrl(url);
-            } catch (InvalidURLException e) {
+                HiveJdbcConnectionOptions params = parseHiveUrl(url);
+                String myEndpoint = String.format("%s-dataproc.googleapis.com:443", params.region());
+
+                // Configure the settings for the cluster controller client.
+                ClusterControllerSettings clusterControllerSettings =
+                        ClusterControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+                ClusterControllerClient clusterControllerClient =
+                        ClusterControllerClient.create(clusterControllerSettings);
+
+                DataprocInfo clusterInfo = new DataprocInfo();
+                clusterInfo.setController(clusterControllerClient);
+                clusterInfo.setParams(params);
+
+                String hiveURL = clusterInfo.toHiveJdbcUrl();
+            } catch (InvalidURLException | IOException e) {
```

If `InvalidURLException` is already an instance of `SQLException`, you don't need to catch this one -- you can just let it bubble up